MSAL/src/native_auth/network/sign_up/MSALNativeAuthSignUpRequestProvider.swift

```diff
@@ -92,7 +89,10 @@ final class MSALNativeAuthSignUpRequestProvider: MSALNativeAuthSignUpRequestProv
         return request
     }
 
-    private func formatAttributes(_ attributes: [String: Any]) throws -> String? {
+    private func formatAttributes(_ attributes: [String: Any]?) throws -> String? {
+        guard let attributes = attributes else {
```

Comment:
Should we strip them also if the developer passes an empty dictionary? As in calls public client interface like this?
`signUpUsingPassword(username: "Test", password: "Pass", attributes: [:], delegate: self)` 
That would mean doing `guard let attributes = attributes, !attributes.isEmpty else {` here

Reply:
Hey Silviu, I took in consideration this approach. Please check the PR description to know more